midend/README.md

```diff
@@ -11,11 +11,11 @@ Refer to the full page here: [Midend](https://p4lang.github.io/p4c/midend.html)
 \internal
 -->
 # Midend
-
-This folder contains passes which may be useful for implementing various mid-ends.
-These are linked as part of the front-end library.
-These passes are not used in the front-end, but are still largely architecture-independent.
-
 <!--!
 \endinternal
 -->
+The [`midend` directory](https://github.com/p4lang/p4c/tree/main/midend) contains passes which may be useful for implementing various mid-ends.
+These are linked as part of the front-end library.
+These passes are not used in the front-end, but are still largely architecture-independent.
+
+The mid-end stage applies further optimizations to the IR of the P4 program. It's architecture-independent, it is driven by policies specific to the target architecture. The mid-end uses the same base IR as the front-end.
```

Comment:
Remove this paragraph, it doesn't add any new information compared to the previous paragraph.